networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/PeerSubnetSubscriptions.java

```diff
@@ -138,18 +137,28 @@ public PeerScorer createScorer() {
   }
 
   public int getSubscribersRequired() {
-    return getMinSubscriberCount().map(c -> Math.max(targetSubnetSubscriberCount - c, 0)).orElse(0);
+    OptionalInt count = getMinSubscriberCount();
+    if (count.isPresent()) {
+      return Math.max(targetSubnetSubscriberCount - count.getAsInt(), 0);
+    } else {
+      return 0;
+    }
   }
 
-  private Optional<Integer> getMinSubscriberCount() {
-    final Optional<Integer> minAttestationSubscribers =
+  private OptionalInt getMinSubscriberCount() {
```

Comment:
Gotcha. Will revert.

Reply:
Hmm this is actually a little harder than I thought it would be. `streamRelevantSubnets()` returns an `IntStream` because `relevantSubnets` is an `IntSet`. As a result, `min()` will return an `OptionalInt`. Should I try to convert it to an `Optional<Integer>`?

Reply:
ah ok, stick with `OptionalInt` in that case. Just a shame they didn't put a `.map` method on it...